src/Symfony/Component/Console/Application.php

```diff
@@ -117,10 +117,17 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         $this->configureIO($input, $output);
 
         try {
+            $e = null;
             $exitCode = $this->doRun($input, $output);
-        } catch (\Exception $e) {
+        } catch (\Exception $x) {
+            $e = $x;
+        } catch (\Throwable $x) {
```

Comment:
Hm, then something else in this commit is causing the behaviour change in PhpSpec reported here https://github.com/symfony/symfony/issues/22678